backend/de.metas.business/src/main/java/de/metas/project/process/C_WO_Project_ObjectUnderTest_OrderProvision.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2023 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.project.process;
+
+import de.metas.bpartner.BPartnerId;
+import de.metas.i18n.AdMessageKey;
+import de.metas.organization.IOrgDAO;
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.IProcessPreconditionsContext;
+import de.metas.process.JavaProcess;
+import de.metas.process.Param;
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.project.ProjectId;
+import de.metas.project.workorder.undertest.WOProjectObjectUnderTest;
+import de.metas.project.workorder.undertest.WorkOrderProjectObjectUnderTestService;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.compiere.SpringContextHolder;
+
+import java.sql.Timestamp;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public class C_WO_Project_ObjectUnderTest_OrderProvision extends JavaProcess implements IProcessPrecondition
+{
+	private static final String WO_ObjectUnderTest_Without_Product_Message = "C_Project_WO_ObjectUnderTest";
+
+	final WorkOrderProjectObjectUnderTestService woProjectObjectUnderTestService = SpringContextHolder.instance.getBean(WorkOrderProjectObjectUnderTestService.class);
+
+	final IOrgDAO orgDAO = Services.get(IOrgDAO.class);
+
+	private static final String PARAM_C_BPARTNER_ID = "C_BPartner_ID";
+	@Param(parameterName = PARAM_C_BPARTNER_ID)
+	private int bPartnerId;
+
+	private static final String PARAM_DATE_PROMISED = "DatePromised";
+	@Param(parameterName = PARAM_DATE_PROMISED)
+	private Timestamp datePromised;
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		final List<WOProjectObjectUnderTest> woProjectObjectUnderTestList = woProjectObjectUnderTestService.getByProjectId(ProjectId.ofRepoId(getRecord_ID()));
+
+		if (!woProjectObjectUnderTestList.isEmpty())
```

Comment:
if no `WOProjectObjectUnderTest` is available, there should be an error thrown as the process should not be available in that scenario